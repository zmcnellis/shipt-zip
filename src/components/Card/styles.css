/* Cards ---------------------------------------------------------------------*/
.cards {
  max-width: 960px;
  margin: 0 auto 30px;
}

/* article {
  position: relative;
} */

.article-img {
  height: 200px;
  width: 100%;
  object-fit: cover;
  transition: all .2s ease-in-out;
}

.article-img:hover {
  transform: scale(1.05);
}

.article-title {
  text-align: center;
  font-weight: bold;
  font-size: 25px;
  line-height: 24px;
  height: 48px;
  overflow: hidden;
}

.article-description {
  text-align: center;
  font-size: 16px;
}
