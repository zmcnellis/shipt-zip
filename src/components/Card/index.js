import React, { Component } from 'react';
import PropTypes from 'prop-types';
import step1 from '../../images/step1.png';
import step2 from '../../images/step2.png';
import step3 from '../../images/step3.png';
import './styles.css';

class Card extends Component {
  render() {
    let imgSrc;
    switch(Number.parseInt(this.props.step)) {
      case 1:
        imgSrc = step1;
        break;
      case 2:
        imgSrc = step2;
        break;
      case 3:
        imgSrc = step3;
        break;
      default:
        imgSrc = step1;
        break;
    }

    return (
      <article>
        <img className="article-img" src={imgSrc} />
        <h1 className="article-title">{this.props.title}</h1>
        <p className="article-description">{this.props.description}</p>
      </article>
    );
  }
}

Card.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  step: PropTypes.string.isRequired
};

export default Card;
