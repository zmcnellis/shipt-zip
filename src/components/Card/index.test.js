import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Card from '.';

describe('Card', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Card title="mockTitle" description="mockDescription" step="1" />
    );
    expect(output).toMatchSnapshot();
  });

  it('should display the correct title', () => {
    const output = shallow(
      <Card title="mockTitle" description="mockDescription" step="1" />
    );
    expect(output.find('.article-title').text()).toEqual('mockTitle');
  });

  it('should display the correct description', () => {
    const output = shallow(
      <Card title="mockTitle" description="mockDescription" step="1" />
    );
    expect(output.find('.article-description').text()).toEqual('mockDescription');
  });
});
