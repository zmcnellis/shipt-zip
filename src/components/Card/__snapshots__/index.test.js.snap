// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    description="mockDescription"
    step="1"
    title="mockTitle"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="mockTitle"
          className="article-img"
          src="step1.png"
/>,
        <h1
          className="article-title"
>
          mockTitle
</h1>,
        <p
          className="article-description"
>
          mockDescription
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "mockTitle",
          "className": "article-img",
          "src": "step1.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "mockTitle",
          "className": "article-title",
        },
        "ref": null,
        "rendered": "mockTitle",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "mockDescription",
          "className": "article-description",
        },
        "ref": null,
        "rendered": "mockDescription",
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="mockTitle"
            className="article-img"
            src="step1.png"
/>,
          <h1
            className="article-title"
>
            mockTitle
</h1>,
          <p
            className="article-description"
>
            mockDescription
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "mockTitle",
            "className": "article-img",
            "src": "step1.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mockTitle",
            "className": "article-title",
          },
          "ref": null,
          "rendered": "mockTitle",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mockDescription",
            "className": "article-description",
          },
          "ref": null,
          "rendered": "mockDescription",
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
